versioned/tiered/rocksdb/src/main/java/com/dremio/nessie/versioned/store/rocksdb/Function.java

```diff
@@ -30,13 +32,32 @@
 
   /**
    * Compares for equality with a provided Function object.
-   * @param function object to compare
+   * @param object  the object to compare
    * @return true if this is equal to provided object
    */
-  boolean equals(Function function) {
-    return (this.getOperator().equals(function.getOperator())
-        && this.getPath().equals(function.getPath())
-        && this.getValue().equals(function.getValue()));
+  @Override
+  public boolean equals(Object object) {
+    if (object == null) {
+      return false;
+    }
+
+    if (object == this) {
+      return true;
+    }
+
+    if (!(object instanceof Function)) {
+      return false;
+    }
+
+    Function function = (Function) object;
```

Comment:
Final - please ensure this is done in the future as we've discussed before.

Reply:
Sorry, updated. Haven't taken a break